databricks/koalas/frame.py

```diff
@@ -3987,6 +3988,71 @@ def describe(self, percentiles: Optional[List[float]] = None) -> 'DataFrame':
                                   index_map=[('summary', None)])
         return DataFrame(internal).astype('float64')
 
+    # TODO: add 'axis' parameter
+    def cummax(self, skipna: bool = True):
+        """
+        Return cumulative maximum over a DataFrame or Series axis.
+
+        Returns a DataFrame or Series of the same size containing the cumulative maximum.
+
+        Parameters
+        ----------
+        skipna : boolean, default True
+            Exclude NA/null values. If an entire row/column is NA, the result will be NA.
+
+        Returns
+        -------
+        Series or DataFrame
+
+        See Also
+        --------
+        DataFrame.max : Return the maximum over DataFrame axis.
+        DataFrame.cummax : Return cumulative maximum over DataFrame axis.
+        DataFrame.cummin : Return cumulative minimum over DataFrame axis.
+        DataFrame.cumsum : Return cumulative sum over DataFrame axis.
+        DataFrame.cumprod : Return cumulative product over DataFrame axis.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame([[2.0, 1.0],
+        ...                    [3.0, None],
+        ...                    [1.0, 0.0]],
+        ...                    columns=list('AB'))
+        >>> df
+             A    B
+        0  2.0  1.0
+        1  3.0  NaN
+        2  1.0  0.0
+
+        By default, iterates over rows and finds the sum in each column.
+
+        >>> df.cummax()
+             A    B
+        0  2.0  1.0
+        1  3.0  NaN
+        2  3.0  1.0
+        """
+        index_columns = self._internal.index_columns
+        data_columns = self._internal.data_columns
+        window = Window.orderBy(index_columns).rangeBetween(Window.unboundedPreceding, 0)
```

Comment:
Seems this code is repeated in your PRs. can we deduplicate? for instance define one function and reuse it.